pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -550,6 +520,29 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication {
         emit AuthorizerChanged(newAuthorizer);
     }
 
+    function _ensureAuthenticatedByRole(address pool, address roleAddress, bool onlyOwner) private view {
```

Comment:
I like the change, but the core logic is still the same.
This is not a strong opinion, but I'd like to hear your thoughts.

How about splitting this one in 2 functions, removing `onlyOwner`?
- For functions that are `onlyOwner`, same name applies; just check whether role address matches `msg.sender` in case there's a role address set.
- For functions that are not only owner, check both role address and call `_canPerform` (`_ensureAuthenticatedByRoleOrAuthorized`? too long?)

You might need a second lower level helper equivalent to `_canPerform` that returns a boolean and doesn't revert for roles.

Reply:
Ok; refactored again to simplify it further.